aiven/resource_static_ip.go

```diff
@@ -0,0 +1,165 @@
+	}
+	for i := range r.StaticIPs {
+		sip := r.StaticIPs[i]
+		if sip.StaticIPAddressID == staticIPAddressId {
+			if err := d.Set("project", project); err != nil {
+				return diag.Errorf("error setting static ips `project` for resource %s: %s", d.Id(), err)
+			}
+			if err := d.Set("cloud_name", sip.CloudName); err != nil {
+				return diag.Errorf("error setting static ips `cloud_name` for resource %s: %s", d.Id(), err)
+			}
+			if err := d.Set("ip_address", sip.IPAddress); err != nil {
+				return diag.Errorf("error setting static ips `ip_address` for resource %s: %s", d.Id(), err)
+			}
+			if err := d.Set("service_name", sip.ServiceName); err != nil {
+				return diag.Errorf("error setting static ips `service_name` for resource %s: %s", d.Id(), err)
+			}
+			if err := d.Set("state", sip.State); err != nil {
+				return diag.Errorf("error setting static ips `state` for resource %s: %s", d.Id(), err)
+			}
+			if err := d.Set("static_ip_address_id", sip.StaticIPAddressID); err != nil {
+				return diag.Errorf("error setting static ips `static_ip_address_id` for resource %s: %s", d.Id(), err)
+			}
+			return nil
+		}
+	}
+	d.SetId("")
+	return nil
+}
+
+func resourceStaticIPCreate(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project := d.Get("project").(string)
+	cloudName := d.Get("cloud_name").(string)
+
+	r, err := client.StaticIPs.Create(project, aiven.CreateStaticIPRequest{CloudName: cloudName})
+	if err != nil {
+		return diag.FromErr(fmt.Errorf("unable to create static ip: %w", err))
+	}
+
+	d.SetId(schemautil.BuildResourceID(project, r.StaticIPAddressID))
+
+	if err := resourceStaticIPWait(ctx, d, m); err != nil {
+		return diag.FromErr(fmt.Errorf("unable to wait for static ip to become active: %w", err))
+	}
+
+	return resourceStaticIPRead(ctx, d, m)
+}
+
+func resourceStaticIPDelete(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	client := m.(*aiven.Client)
+
+	project, staticIPAddressId := schemautil.SplitResourceID2(d.Id())
+
+	err := client.StaticIPs.Delete(
+		project,
+		aiven.DeleteStaticIPRequest{
+			StaticIPAddressID: staticIPAddressId,
+		})
+	if err != nil && !aiven.IsNotFound(err) {
+		return diag.FromErr(err)
+	}
+	return nil
+}
+
+func resourceStaticIPWait(ctx context.Context, d *schema.ResourceData, m interface{}) error {
+	client := m.(*aiven.Client)
+
+	project, staticIPAddressId := schemautil.SplitResourceID2(d.Id())
+
+	conf := resource.StateChangeConf{
+		Target:  []string{"created"},
+		Pending: []string{"creating", "waiting"},
+		Timeout: d.Timeout(schema.TimeoutCreate),
+		Refresh: func() (result interface{}, state string, err error) {
+			log.Println("[DEBUG] checking if static ip", staticIPAddressId, "is in 'created' state")
+			r, err := client.StaticIPs.List(project)
+			if err != nil {
+				return nil, "", fmt.Errorf("unable to fetch static ips: %w", err)
+			}
+			for i := range r.StaticIPs {
```

Comment:
same here I think, we can probably replace it with `_,sip := ...`

Reply:
:heavy_check_mark: